tests/test_form.py

```diff
@@ -123,6 +123,27 @@ def test_choose_submit_from_selector(value):
     assert res.status_code == 200 and res.text == 'Success!'
 
 
+@pytest.mark.parametrize("value", [
+    pytest.param('continue', id='first'),
+    pytest.param('cancel', id='second'),
+])
+def test_choose_submit_from_selector_image(value):
+    """Test choose_submit by passing a CSS selector argument."""
+    text = """
+    <form method="post" action="mock://form.com/post">
+      <input type="image" name="do" value="continue" />
+      <input type="image" name="do" value="cancel" />
```

Comment:
> `<input type="image">` elements do not accept value attributes. The path to the image to be displayed is specified in the src attribute.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/image

While this obviously this impacts the test (which can be easily modified), it's not clear to me whether or not it impacts the implementation of the fix. I don't think it does, but it's something to think twice about.

Reply:
I think I got what u mean. This test could be an input type submit and another input type image and making sure that the input image would be picked - with different names, of course. What do you think?

Reply:
Ultimately, we need to try to replicate what the browser does (especially in cases where this is the standard). When clicking on a `<input type="image" name="test" ...>`, the browser does not submit a `name=value` request, because `value` is an invalid attribute for `type="image"`. Instead, it will submit a request that looks like:
```
test.x=171&test.y=168 
```
where 171 and 168 are the mouse positions where the image was clicked. Obviously we don't have access to mouse positions with MechanicalSoup, so I'm not exactly sure how we _should_ handle this, but I know we don't want to try to use a `value` in this case (or at the very least, we should allow for the case where `value` doesn't exist, and just submit a `name=` request). Or maybe we should submit `name.x=0&name.y=0`?

@moy Do you have any thoughts about what the request should look like?

Reply:
I think `name.x=0&name.y=0` by default makes sense. Ideally, there would be a way to specify `x` and `y` from MechanicalSoup, but this can wait for another PR.